src/AspNetCore.Lambda/HttpHandlers.fs

```diff
@@ -275,4 +291,25 @@ let htmlFile (relativeFilePath : string) =
                 ctx
                 |> (setHttpHeader "Content-Type" "text/html"
                 >>= setBodyAsString html)
-        }
\ No newline at end of file
+        }
+
+let private handlerWithRootedPath (path:string) (handler : HttpHandler) = 
+    fun (ctx : HttpHandlerContext) ->
+        async {
+            let savedSubPath = getSavedSubPath ctx.HttpContext
+            try
+                ctx.HttpContext.Items.Item RouteKey <- ((savedSubPath |> Option.defaultValue "") + path)
+                return! handler ctx
+            finally
+                match savedSubPath with
+                | Some savedSubPath -> ctx.HttpContext.Items.Item RouteKey <- savedSubPath
+                | None              -> ctx.HttpContext.Items.Remove RouteKey |> ignore
+        }
+
```

Comment:
I'm in no hurry. API are easy to add but then you are stuck with them forever. That said, I think I'm happy with it in it's current state.